Add keithmattix as a networking maintainer

keithmattix opened the pull request:
Here are some of the PRs I've completed in this area:
istio/istio#45925
istio/istio#45593
istio/istio#45453
istio/istio.io#13385
istio/istio#44894
istio/ztunnel#498
istio/istio#43497

ericvn commented:
LGTM

hzxuzhonghu commented:
Thanks for all the contributions, i can see most of the work are around ambient peerAuthentication, but istio networking is more wide, mostly covering sidecar & ambient xds discovery, xds generator, etc.

To me they are not equal, FYI: https://github.com/istio/community/blob/master/ROLES.md#requirements-3

howardjohn commented:
I believe this gets voted on by networking WG leads, which I am not one of, but fwiw I support this.

> Thanks for all the contributions, i can see most of the work are around ambient peerAuthentication, but istio networking is more wide, mostly covering sidecar & ambient xds discovery, xds generator, etc.

One note is there is a "Pilot" OWNERS group which handles some of these areas; this PR is not adding to that group

ramaraochavali commented:
@keithmattix Thank you for all the work. I think your design docs and your contributions interms of reviews are good. I kind of agree with @hzxuzhonghu 's feedback that the work you have done till now is limited to very specific functionality mostly(peer authentication) whereas networking (even without pilot) is broader than that.  Also looking at the [requirements](https://github.com/istio/community/blob/master/ROLES.md#requirements-3), we need to have ~30 substantial PRs and you have **total** of 13 [merged](https://github.com/istio/istio/pulls?q=is%3Apr+author%3Akeithmattix+is%3Amerged+) PRs.  

I agree with the rest of your assessment though. Can you please work on filling this gap in PRs (in the broader networking codebase) and we can revisit this in a month or two?

ramaraochavali commented:
I think you are right about number of PRs - It is across Istio repo (Not just Istio/Istio).  As discussed offline, please focus on sidecar networking + gateway a bit more as these will be heavily used while Ambient matures to beta+. We can definitely circle back in a Month and I am sure based on your contributions till now, you will be great addition. Thank you.

ericvn commented:
> I believe this gets voted on by networking WG leads, which I am not one of, but fwiw I support this.

As I understand it, the WG leads have to approve, but the other WG leads also approve (or at least not object). My comment above was just my non-objection.

ramaraochavali commented:
Can you please update here with recent PRs that you have done in networking area?

keithmattix commented:
@ramaraochavali Sure! I fixed a Gateway [bug](https://github.com/istio/istio/pull/46706) with our samples that blocked the 1.19 release, fixed HTTPS [service entries ](https://github.com/istio/istio/pull/46795) in dualstack mode, and added [tcp keepalives](https://github.com/istio/ztunnel/pull/662). I also have networking 3 PRs in review: 

istio/istio#46833 - required an upstream gateway-api conformance [bugfix](https://github.com/kubernetes-sigs/gateway-api/pull/2375)
istio/istio#46915 - brings Istio's requestauthentication resource more in line with the gateway api community's policy attachment specification
istio/istio#46983 - does the same as above but for WASMPlugin

kfaseela reviewed (approved):
Congratulations 🙂
